ext/ddtrace_profiling_native_extension/stack_recorder.c

```diff
@@ -386,8 +387,9 @@ static VALUE _native_serialize(DDTRACE_UNUSED VALUE _self, VALUE recorder_instan
   VALUE start = ruby_time_from(ddprof_start);
   VALUE finish = ruby_time_from(ddprof_finish);
 
-  if (!ddog_prof_Profile_reset(args.profile, NULL /* start_time is optional */ )) {
-    return rb_ary_new_from_args(2, error_symbol, rb_str_new_cstr("Failed to reset profile"));
+  ddog_prof_Profile_Result reset_result = ddog_prof_Profile_reset(args.profile, NULL /* start_time is optional */ );
+  if (reset_result.tag == DDOG_PROF_PROFILE_RESULT_ERR) {
+    return rb_ary_new_from_args(2, error_symbol, rb_sprintf("Failed to reset profile: %"PRIsVALUE, get_error_details_and_drop(&reset_result.err)));
   }
```

Comment:
Could we use `reset_profile` here? 

Maybe the function `reset_profile` could return if the profiler successfully rest and if not the caller can choose what to do

Reply:
Simplified as 80d8cecf27988952a8cc180655fd17788629a64c